core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java

```diff
@@ -6677,12 +6677,37 @@ private void checkLiteral2(String expression, String expected) {
    * Add BITAND_AGG, BITOR_AGG functions (enabled in Snowflake library)</a>. */
   @Test void testBitOrAgg() {
     final String query = "select bit_or(\"product_id\")\n"
-        + "from \"product\"";
+      	+ "from \"product\"";
     final String expectedSnowflake = "SELECT BITOR_AGG(\"product_id\")\n"
-        + "FROM \"foodmart\".\"product\"";
+      	+ "FROM \"foodmart\".\"product\"";
     sql(query).withLibrary(SqlLibrary.SNOWFLAKE).withSnowflake().ok(expectedSnowflake);
   }
 
+  /** Test case for
+   * <a href="https://issues.apache.org/jira/browse/CALCITE-6220">[CALCITE-6220]
+   * Rewrite MIN/MAX(bool) as BOOL_AND/BOOL_OR for Postgres, Redshift</a>. */
+  @Test void testMaxMinOnBooleanColumn() {
+    final String query = "select max(\"brand_name\" = 'a'), "
+        + "min(\"brand_name\" = 'a'), "
+        + "min(\"brand_name\")\n"
+        + "from \"product\"";
+    final String expected = "SELECT MAX(\"brand_name\" = 'a'), "
+        + "MIN(\"brand_name\" = 'a'), "
+        + "MIN(\"brand_name\")\n"
+        + "FROM \"foodmart\".\"product\"";
+    final String expectedPostgres = "SELECT BOOL_OR(\"brand_name\" = 'a'), "
+        + "BOOL_AND(\"brand_name\" = 'a'), "
+        + "MIN(\"brand_name\")\n"
+        + "FROM \"foodmart\".\"product\"";
+    final String expectedRedshift = "SELECT BOOL_OR(\"brand_name\" = 'a'), "
+        + "BOOL_AND(\"brand_name\" = 'a'), "
+        + "MIN(\"brand_name\")\n"
+        + "FROM \"foodmart\".\"product\"";
+    sql(query).ok(expected);
+    sql(query).withPostgresql().ok(expectedPostgres);
```

Comment:
I would chain. And add a test for a 'neutral' dialect:
```
 sql(query).ok(expected)
    .withPostgresql().ok(expectedPostgres)
    .withRedshift().ok(expectedRedshift)
    .withMySQL().ok(expected);
```

I like how you tested on a non-BOOLEAN column

Reply:
Done